src/overlays/actors/ovl_En_Rg/z_en_rg.c

```diff
@@ -84,75 +111,766 @@ static DamageTable D_80BF57C4 = {
+        if (this->unk_18C->id == ACTOR_PLAYER) {
+            D_80BF5C10 = 0;
+        }
+        this->unk_18C = NULL;
+        this->unk_310 &= ~0x800;
+    }
+
+    if (this->unk_310 & 0x20) {
+        if (this->collider2.base.at != NULL) {
+            if (this->collider2.base.at->id == ACTOR_OBJ_TSUBO) {
+                this->collider2.base.at->params |= 0x3F;
+                this->unk_320 += 0x190;
+                this->unk_320 = CLAMP_MAX(this->unk_320, 0x190);
+            } else if (this->collider2.base.at->id == ACTOR_PLAYER) {
+                this->unk_326 = 0x28;
+                if (player->stateFlags3 & 0x1000) {
+                    player->linearVelocity *= 0.5f;
+                    player->unk_B08[0] = player->linearVelocity;
+                    player->unk_B08[1] += player->linearVelocity * 0.05f;
+                    if (BINANG_SUB(this->actor.yawTowardsPlayer, player->currentYaw) > 0) {
+                        player->currentYaw += 0x2000;
+                    } else {
+                        player->currentYaw -= 0x2000;
+                    }
+                    player->unk_B8C = 4;
+                    player->invincibilityTimer = 20;
+                    player->actor.home.rot.y = player->actor.world.rot.y;
+                    player->actor.shape.rot.y = player->actor.world.rot.y;
+                }
+
+                if ((this->unk_18C != NULL) && (this->unk_18C->id == ACTOR_PLAYER)) {
+                    D_80BF5C10 = 0;
+                }
+            }
+        }
+        this->unk_18C = NULL;
+        this->unk_310 &= ~0x800;
+        this->unk_310 &= ~0x20;
+    }
+}
+
+s32 func_80BF4D64(Vec3f* arg0) {
+    return Math3D_XZBoundCheck(-1261.0f, -901.0f, -1600.0f, -1520.0f, arg0->x, arg0->z);
+}
+
+s32 func_80BF4DA8(EnRg* this) {
+    static Vec3f D_80BF596C[] = {
+        { -2473.0f, 39.0f, 7318.0f },  { -2223.0f, 142.0f, 7184.0f }, { -2281.0f, 41.0f, 7718.0f },
+        { -2136.0f, 96.0f, 7840.0f },  { -2432.0f, 6.0f, 7857.0f },   { -2412.0f, 139.0f, 6872.0f },
+        { -2719.0f, 39.0f, 7110.0f },  { -2289.0f, 67.0f, 7463.0f },  { -2820.0f, 85.0f, 6605.0f },
+        { -2088.0f, 160.0f, 7584.0f }, { -2503.0f, 1.0f, 7643.0f },
+    };
+    s32 pad[4];
+    s32 ret = false;
+    f32 temp_f20;
+    s16 temp_v1;
+    s32 i;
+
+    for (i = 0; i < ARRAY_COUNT(D_80BF596C); i++) {
+        temp_f20 = Actor_XZDistanceToPoint(&this->actor, &D_80BF596C[i]);
+        temp_v1 = Actor_YawToPoint(&this->actor, &D_80BF596C[i]) - this->actor.world.rot.y;
+
+        if ((temp_f20 < 100.0f) && (ABS_ALT(temp_v1) < 0xC00)) {
+            if (temp_v1 > 0) {
+                this->actor.world.rot.y -= 0x1600;
+            } else {
+                this->actor.world.rot.y += 0x1600;
+            }
+
+            this->unk_324 = 20;
+            this->unk_310 &= ~0x800;
+
+            if ((this->unk_18C != ACTOR_PLAYER) && (this->unk_18C->id == ACTOR_PLAYER)) {
```

Comment:
unk_18C is an `Actor` pointer, so the comparison here should be a NULL comparison instead.
```suggestion
            if ((this->unk_18C != NULL) && (this->unk_18C->id == ACTOR_PLAYER)) {
```